src/System.Net.Security/tests/FunctionalTests/ClientAsyncAuthenticateTest.cs

```diff
@@ -43,7 +43,8 @@ public async Task ClientAsyncAuthenticate_ServerRequireEncryption_ConnectWithEnc
         [Fact]
         public async Task ClientAsyncAuthenticate_ServerNoEncryption_NoConnect()
         {
-            await Assert.ThrowsAsync<IOException>(() => ClientAsyncSslHelper(EncryptionPolicy.NoEncryption));
+            // Don't use Tls13  since we are trying to use NullEncryption
+            await Assert.ThrowsAsync(TestConfiguration.SupportsHandshakeAlerts && TestConfiguration.SupportsNullEncryption ? typeof(AuthenticationException) : typeof(IOException),() => ClientAsyncSslHelper(EncryptionPolicy.NoEncryption, SslProtocolSupport.DefaultSslProtocols,  SslProtocols.Tls | SslProtocols.Tls11 |  SslProtocols.Tls12 ));
```

Comment:
This line would really like some linebreaks in it.  Maybe something like
```suggestion
            Type expectedExceptionType = TestConfiguration.SupportsHandshakeAlerts && TestConfiguration.SupportsNullEncryption ?
                typeof(AuthenticationException) :
                typeof(IOException);
                 
            await Assert.ThrowsAsync(
                expectedExceptionType,
                () => ClientAsyncSslHelper(
                    EncryptionPolicy.NoEncryption,
                    SslProtocolSupport.DefaultSslProtocols,
                    SslProtocols.Tls | SslProtocols.Tls11 |  SslProtocols.Tls12 ));
```

Reply:
agreed. and updated.